src/App.tsx

```diff
@@ -26,70 +27,145 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((good1: string, good2: string) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      case SortType.ALPABET:
+        return good1.localeCompare(good2);
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: 0,
+  };
+
+  handleSortClick = (
+    currentTarget: EventTarget & HTMLButtonElement,
+  ) => {
+    const { value } = currentTarget;
+
+    this.setState({ sortType: +value });
+  };
+
+  handleResetClick = (
+    currentTarget: EventTarget & HTMLButtonElement,
+  ) => {
+    this.handleSortClick(currentTarget);
+    this.setState({ isReversed: false });
+  };
+
+  handleReverseClick = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  render() {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            value="1"
+            className={classNames(
+              'button',
+              'is-info',
+              {
+                'is-light': this.state.sortType !== SortType.ALPABET,
+              },
+            )}
+            onClick={(event) => {
+              this.handleSortClick(event.currentTarget);
```

Comment:
```suggestion
              this.handleSortClick(SortType.ALPHABET);
```
Пропоную сотрувати не по номерам currentTarget.value, а задати в enum SortType строкові значення і використати їх

Reply:
так буде більш наглядно по коду